src/iframeMessenger.js

```diff
@@ -423,6 +423,31 @@
             }
         }
 
+        function _addAcquisitionDataToLinks(acquisitionData) {
+            var links = document.getElementsByClassName('js-acquisition-link');
+            var json = encodeURIComponent(JSON.stringify(acquisitionData));
+            for (var i = 0; i < links.length; i++) {
+                var href = link.attr('href');
+                if (href) {
+                    href = `${href}${href.includes('?') ? '&' : '?'}${json}`;
+                    link.attr('href', href);                    
+                }
+            }
+        }
+
+        function _enrichAcquisitionLinks(acquisitionData) {
+            _addAcquisitionDataToLinks(acquisitionData);
+            var message = { type: 'enrich-acquisition-links' };
+            _postMessage(message, function(data) {
+                var referrerData = data.referrerData;
+                if (referrerData && data.type === message.type) {
+                    acquisitionData.referrerUrl = referrerData.referrerUrl;
```

Comment:
add discussed - it would be nice to not mutate this object as the caller might not realise this is happening